### PR TITLE
feat(web): include `includeNullValues` prop to handle sparse data

### LINKED ISSUE #1059: Do not apply pristine range filters
**Affected Projects**
React

**Library Version:** x.y.z
3.0.0-rc.20

**Describe the bug**
When filtering sparse data (i.e., when some of the attributes can be present, but may not be), you want filter components to only be applied when the user explicitly sets them.

For example: I work on a search app that filters by some date field, but that date field is only present in 40% of the documents. When I render the `RangeSlider` component, only items are shown that contain some value for the `dataField` attribute. This means that only documents are shown that have that specific date field.

This happens, because the `RangeSlider` applies a filter to the query with the `range` `start` and `end` values.

Ideally, this filter is only applied after the user interacts with the component. This could be fixed by adding a `pristine` state to `RangeSlider` and others, and only applies the filter when its value is false.

Perhaps this could be set as an optional prop, e.g. `ignoreUntouched`.

### PR DESCRIPTION
**Pull request by ShahAnuj2610**
fixes #1059 #1096 
- feat - add support in all range components for the prop `includeNullValues`

### CODE REVIEW

File: packages/web/src/components/range/RangeInput.js [hunk 1]
```diff
@@ -185,6 +185,7 @@ RangeInput.propTypes = {
 	style: types.style,
 	themePreset: types.themePreset,
 	componentId: types.stringRequired,
+	includeNullValues: types.bool,
```
Comment (bietkul) [inline packages/web/src/components/range/RangeInput.js:188]: @ShahAnuj2610 Are we missing the query generation here?

File: packages/web/src/components/range/RatingsFilter.js [hunk 2]
```diff
@@ -134,13 +139,13 @@ class RatingsFilter extends Component {
 				},
 			};
 			if (includeNullValues) {
```
Comment (bietkul) [inline packages/web/src/components/range/RatingsFilter.js]: I think we're already checking it in the util function.

**Review by bietkul — CHANGES_REQUESTED** [review]
Typings are missing for `includeNullValues` in `.ts` files for all the components.

File: packages/web/src/components/range/RangeInput.js (hunk 1 above)
Comment (ShahAnuj2610) [reply packages/web/src/components/range/RangeInput.js:188]: can you elaborate?

Comment (bietkul) [reply packages/web/src/components/range/RangeInput.js:188]: Ok, I think it is not needed.

File: packages/web/src/components/range/RatingsFilter.js [hunk 3]
```diff
@@ -129,23 +124,10 @@ class RatingsFilter extends Component {
 	static defaultQuery = (value, props, includeNullValues = false) => {
```
Comment (bietkul) [inline packages/web/src/components/range/RatingsFilter.js]: @ShahAnuj2610 Why `includeNullValues` is the third argument?
We can access it from `props`.

**Review by bietkul — CHANGES_REQUESTED** [review]

File: packages/web/src/components/range/RatingsFilter.js (hunk 3 above)
Comment (ShahAnuj2610) [reply packages/web/src/components/range/RatingsFilter.js]: no, the third argument comes from the `includeUnrated` in the data object
`
			const handleUpdates = () => {
				this.updateQuery(value, props, includeUnrated);
				this.locked = false;
				if (props.onValueChange) props.onValueChange(value);
			}
`

Comment (ShahAnuj2610) [reply packages/web/src/components/range/RatingsFilter.js]: https://github.com/appbaseio/reactivesearch/pull/1003

Comment (bietkul) [reply packages/web/src/components/range/RatingsFilter.js]: But why you named it `includeNullValues`?

**Review by bietkul — CHANGES_REQUESTED** [review]

**Comment by ShahAnuj2610** [comment]
@bietkul can rename `includeNullValues` param in func to `includeUnrated`

File: packages/web/src/components/range/RatingsFilter.js [hunk 4]
```diff
@@ -121,12 +121,12 @@ class RatingsFilter extends Component {
...
 	};
 
-	static defaultQuery = (value, props, includeNullValues = false) => {
+	static defaultQuery = (value, props, includeUnrated = false) => {
 		let query = null;
 		if (value) {
 			query = getRangeQueryWithNullValues(value, {
 				dataField: props.dataField,
-				includeNullValues: props.includeNullValues || includeNullValues,
+				includeNullValues: props.includeNullValues || includeUnrated,
```
Comment (bietkul) [inline packages/web/src/components/range/RatingsFilter.js:129]: @ShahAnuj2610 I'm just curious to know that how these(`includeNullValues` & `includeUnrated`) props are related to each other?

**Review by bietkul — CHANGES_REQUESTED** [review]

File: packages/web/src/components/range/RatingsFilter.js (hunk 4 above)
Comment (ShahAnuj2610) [reply packages/web/src/components/range/RatingsFilter.js:129]: the query executed for both these parameters are same, only `includeUnrated` is applied at the item level, whereas `includeNullValues` is applied for every item in the data
`
<RatingsFilter
					componentId="RatingsSensor"
					dataField="average_rating_rounded"
					title="RatingsFilter"
					data={[
						{ start: 4, end: 5, label: '4 stars and up', includeUnrated: true },
						{ start: 3, end: 5, label: '3 stars and up' },
						{ start: 2, end: 5, label: '2 stars and up' },
						{ start: 1, end: 5, label: '> 1 stars' },
					]}
				/>
`
In the above if `includeNullValues=false`, then null results will only be included for the first data item

**Review by bietkul — APPROVED** [review]

**Pull request merged**